jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -228,10 +282,15 @@ def line_visible(state):
             # exclude for scatter layers where the marker is shown instead of the line
             return getattr(state, 'line_visible', True)
 
+        def state_attr_for_line_visible(state):
+            if is_scatter(state):
+                return 'line_visible'
+            return 'visible'
+
         # Profile/line viewer/layer options:
-        self.line_visible = PlotOptionsSyncState(self, self.viewer, self.layer, 'visible',
+        self.line_visible = PlotOptionsSyncState(self, self.viewer, self.layer, state_attr_for_line_visible,  # noqa

```

Comment:
this is what necessitates most of the changes to `template_mixin`, we want to handle two differently-named glue state attributes (`visible` for profile viewers and `line_visible` for scatter viewers) into the same plot options object.